test/src/unittests/tonal/test_pitchcontours.py

```diff
@@ -0,0 +1,216 @@
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.        
+        self.assertAlmostEqual(duration, 0.0058, 8)
+
+    def testZeros(self):
+        bins, saliences, startTimes, duration = PitchContours()(array(zeros([2,256])), array(zeros([2,256])))      
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        # Previous tests showed small duration of 0.0058 seconds for zero or empty inputs.
+        self.assertAlmostEqual(duration, 0.0058, 8)
+
+    def testZerosUnequalInputs(self):        
+        peakBins = [zeros(4096), zeros(4096)]
+        peakSaliences = [zeros(1024), zeros(1024)]
+        self.assertRaises(RuntimeError, lambda: PitchContours()(peakBins, peakSaliences))
+
+    def testOnes(self):
+        peakBins = [ones(4096),ones(4096)]
+        peakSaliences = [ones(4096),ones(4096)]
+        bins, saliences, startTimes, duration = PitchContours()(peakBins, peakSaliences)
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(saliences, [])
+        self.assertEqualVector(startTimes, [])
+        self.assertAlmostEqual(duration, 0.0058, 3)
+        
+    def testOnesUnequalInputs(self):
+        peakBins = [ones(4096), ones(4096)]
+        peakSaliences = [ones(1024), ones(1024)]
+        self.assertRaises(RuntimeError, lambda: PitchContours()(peakBins, peakSaliences))
+
+    def testRegressionSynthetic(self):
+        # Use synthetic audio for Regression Test. This keeps NPY files size low.     
+        sr = 44100
+        frameSize = 1024
+        hopSize = 512
+
+        size = 1*sr
+        sine1 = [sin(2.0*pi*100.0*i/sr) for i in range(size)]
+        sine2 = [sin(2.0*pi*1000.0*i/sr) for i in range(size)]
+        fc1 = FrameCutter()
+        fc2 = FrameCutter()
+        frame1 = fc1(sine1)
+        frame2 = fc2(sine2)
+        audio = frame1+frame2
+
+        psf = PitchSalienceFunction()
+        psfp = PitchSalienceFunctionPeaks()
+        w = Windowing(type='hann', normalized=False)
+        spectrum = Spectrum()
+        spectralpeaks = SpectralPeaks()
+        pc = PitchContours()
+
+        # Populate an array of frame-wise vectors of cent bin values representing each contour
+        peakBins = []
+        # Populate a frame-wise array of values of salience function peaks
+        peakSaliences = []        
+        for frame in FrameGenerator(audio, frameSize=1024, hopSize=hopSize,
+                                    startFromZero=True):
+            freq_peaks, mag_peaks = spectralpeaks(spectrum(w(frame)))
+            len_freq_peaks = len(freq_peaks)
+            len_mag_peaks = len(mag_peaks)
+            
+            # Skip the first vector elements that correspond to zero frequency
+            freq_peaksp1 = freq_peaks[1:len_freq_peaks]
+            # Do the same for mag. vector to keep the lengths the same
+            mag_peaksp1 = mag_peaks[1:len_mag_peaks]
```

Comment:
We don't need to skip those, because we expect the PitchSalienceFunction to ignore peaks below its reference frequency.